public/src/components/switchboard.tsx

```diff
@@ -97,61 +136,164 @@ const Switchboard: React.FC<InnerProps<SupportFrontendSwitches>> = ({
+    group: [string, SwitchGroup],
     isChecked: boolean,
   ): void => {
     const updatedState = cloneDeep(data);
-
+    const [groupId, groupData] = group;
     updatedState[groupId].switches[switchId].state = isChecked ? SwitchState.On : SwitchState.Off;
-
+    const currentSwitchState = updatedState[groupId].switches[switchId].state;
     setData(updatedState);
+    setNewUnsavedData(true);
+    setPendingChange(
+      'Turned ' + currentSwitchState + ':',
+      switchData.description,
+      groupData.description,
+    );
     setNeedToSaveDataWarning(true);
   };
 
   const createSwitchesFromGroupData = (
     switchId: string,
     switchData: Switch,
-    groupId: string,
+    group: [string, SwitchGroup],
   ): JSX.Element => {
     const isChecked = switchData.state === 'On';
 
     return (
-      <FormControlLabel
-        label={switchData.description}
-        checked={switchData.state === 'On'}
-        control={<SwitchUI />}
-        onChange={(): void => updateSwitchSetting(switchId, switchData, groupId, !isChecked)}
-        value={switchId}
-        key={switchId}
-      />
+      <div className={classes.switchLayout}>
+        <FormControlLabel
+          label={switchData.description}
+          checked={switchData.state === 'On'}
+          control={<SwitchUI />}
+          onChange={(): void => updateSwitchSetting(switchId, switchData, group, !isChecked)}
+          value={switchId}
+          key={switchId}
+        />
+        <IconButton
+          onClick={(): void => actionRemoveSwitchData(group, switchId, switchData.description)}
+          aria-label="removeSwitch"
+        >
+          <DeleteIcon />
+        </IconButton>
+      </div>
     );
   };
 
   const createGroupsFromData = (group: [string, SwitchGroup]): JSX.Element => {
     const [groupId, groupData] = group;
+    const { register, handleSubmit, errors } = useForm<FormData>();
+
+    const onSubmit = ({ switchId, description }: FormData): void => {
+      const updatedState = cloneDeep(data);
+
+      updatedState[groupId].switches[switchId] = {
+        description: description,
+        state: SwitchState.Off,
+      };
+      setData(updatedState);
+      setNewUnsavedData(true);
+      setPendingChange('Added', description, groupData.description);
+      setNeedToSaveDataWarning(true);
+    };
 
     return (
       <FormControl className={classes.formControl} key={groupId}>
-        <FormLabel>{groupData.description}</FormLabel>
+        <FormLabel>
+          {' '}
+          <strong>{groupData.description} </strong>
+        </FormLabel>
+        <br />
         {Object.entries(groupData.switches)
           .sort(sortByDescription)
           .map(([switchId, switchData]) =>
-            createSwitchesFromGroupData(switchId, switchData, groupId),
+            createSwitchesFromGroupData(switchId, switchData, group),
           )}
+        <div className={classes.inputGroup}>
+          <TextField
+            className={classes.input}
```

Comment:
At the moment, these text fields don’t have a screen-reader-accessible description (tested by tabbing through the page with VoiceOver turned on). From [the material-ui/core docs](https://mui.com/material-ui/react-text-field/#accessibility), it looks like this is because we haven’t given the text fields IDs. To make unique IDs we can use the switch group ID:

```suggestion
            className={classes.input}
            id={groupId + '-add-switch-switch-name'}
```